File: frontend/lib/models/turn_game.dart
enum GameType { ticTacToe }

class Turn {
  final String player;
  final String turnInfo;

  Turn({required this.player, required this.turnInfo});

  factory Turn.fromJson(Map<String, dynamic> json) {
    return Turn(player: json['player'] ?? '', turnInfo: json['turnInfo'] ?? '');
  }

  Map<String, dynamic> toJson() => {'player': player, 'turnInfo': turnInfo};
}

class TurnGame {
  final String gameId;
  final GameType gameType;
  final List<Turn> turns;
  final String initiator;
  final String currentPlayer;
  final String winner;

  TurnGame({
    required this.gameId,
    required this.gameType,
    required this.turns,
    required this.initiator,
    required this.currentPlayer,
    required this.winner,
  });

  factory TurnGame.fromJson(Map<String, dynamic> json) {
    return TurnGame(
      gameId: json['gameId'] ?? '',
      gameType: GameType.values.firstWhere(
        (e) => e.toString() == 'GameType.${json['gameType']}',
        orElse: () => GameType.ticTacToe,
      ),
      turns: (json['turns'] as List<dynamic>)
          .map((turn) => Turn.fromJson(turn))
          .toList(),
      initiator: json['initiator'] ?? '',
      currentPlayer: json['currentPlayer'] ?? '',
      winner: json['winner'] ?? '',
    );
  }

  Map<String, dynamic> toJson() => {
    'gameId': gameId,
    'gameType': gameType.toString(),
    'turns': turns.map((turn) => turn.toJson()).toList(),
    'initiator': initiator,
    'currentPlayer': currentPlayer,
    'winner': winner,
  };
}

File: backend/models/turnGame.go
package models

type GameType string
const (
	GameTypeTicTacToe GameType = "ticTacToe"
)

type Turn struct {
	Player string `bson:"player" json:"player"`
	TurnInfo string `bson:"turnInfo" json:"turnInfo"`
}

type TurnGame struct {
	GameID string `bson:"gameId" json:"gameId"`
	GameType GameType `bson:"gameType" json:"gameType"`
	Turns []Turn `bson:"turns" json:"turns"`
	Initiator string `bson:"initiator" json:"initiator"`
	CurrentPlayer string `bson:"currentPlayer" json:"currentPlayer"`
	Winner string `bson:"winner" json:"winner"`
}